integration/query_comparision_test.go

```diff
@@ -944,8 +944,166 @@ func TestQueryComparisonLte(t *testing.T) {
+		},
+		"NeEmptyBinary": {
+			q:              bson.D{{"value", bson.D{{"$ne", primitive.Binary{Data: []byte{}}}}}},
+			filterDatatype: "binary-empty",
+		},
 
-// $nin
+		"NeBoolFalse": {
+			q:              bson.D{{"value", bson.D{{"$ne", false}}}},
+			filterDatatype: "bool-false",
+		},
+		"NeBoolTrue": {
+			q:              bson.D{{"value", bson.D{{"$ne", true}}}},
+			filterDatatype: "bool-true",
+		},
+
+		"NeDatetime": {
+			q:              bson.D{{"value", bson.D{{"$ne", primitive.NewDateTimeFromTime(time.Date(2021, 11, 1, 10, 18, 42, 123000000, time.UTC))}}}},
+			filterDatatype: "datetime",
+		},
+		"NeDatetimeEpoch": {
+			q:              bson.D{{"value", bson.D{{"$ne", primitive.NewDateTimeFromTime(time.Unix(0, 0))}}}},
+			filterDatatype: "datetime-epoch",
+		},
+		"NeDatetimeYearMax": {
+			q:              bson.D{{"value", bson.D{{"$ne", primitive.NewDateTimeFromTime(time.Date(0, 1, 1, 0, 0, 0, 0, time.UTC))}}}},
+			filterDatatype: "datetime-year-min",
+		},
+		"NeDatetimeYearMin": {
+			q:              bson.D{{"value", bson.D{{"$ne", primitive.NewDateTimeFromTime(time.Date(9999, 12, 31, 23, 59, 59, 999000000, time.UTC))}}}},
+			filterDatatype: "datetime-year-max",
+		},
+
+		"NeTimestamp": {
+			q:              bson.D{{"value", bson.D{{"$ne", primitive.Timestamp{T: 42, I: 13}}}}},
+			filterDatatype: "timestamp",
+		},
+		"NeTimestampI": {
+			q:              bson.D{{"value", bson.D{{"$ne", primitive.Timestamp{I: 1}}}}},
+			filterDatatype: "timestamp-i",
+		},
+
+		"NeNull": {
+			q:              bson.D{{"value", bson.D{{"$ne", nil}}}},
+			filterDatatype: "null",
+		},
+
+		"NeInt32": {
+			q:              bson.D{{"value", bson.D{{"$ne", int32(42)}}}},
+			filterDatatype: "int32",
+		},
+		"NeInt32Zero": {
+			q:              bson.D{{"value", bson.D{{"$ne", int32(0)}}}},
+			filterDatatype: "int32-zero",
+		},
+		"NeInt32Max": {
+			q:              bson.D{{"value", bson.D{{"$ne", int32(math.MaxInt32)}}}},
+			filterDatatype: "int32-max",
+		},
+		"NeInt32Min": {
+			q:              bson.D{{"value", bson.D{{"$ne", int32(math.MinInt32)}}}},
+			filterDatatype: "int32-min",
+		},
+
+		"NeInt64": {
+			q:              bson.D{{"value", bson.D{{"$ne", int64(42)}}}},
+			filterDatatype: "int64",
+		},
+		"NeInt64Zero": {
+			q:              bson.D{{"value", bson.D{{"$ne", int64(0)}}}},
+			filterDatatype: "int64-zero",
+		},
+		"NeInt64Max": {
+			q:              bson.D{{"value", bson.D{{"$ne", int64(math.MaxInt64)}}}},
+			filterDatatype: "int64-max",
+		},
+		"NeInt64Min": {
+			q:              bson.D{{"value", bson.D{{"$ne", int64(math.MinInt64)}}}},
+			filterDatatype: "int64-min",
+		},
+
+		"Regex": {
+			q: bson.D{{"value", bson.D{{"$ne", primitive.Regex{Pattern: "foo"}}}}},
+			err: mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: `Can't have regex as arg to $ne.`,
+			},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var actual []bson.D
```

Comment:
here we use to move actual to the line 1101, and add before that line a blank one.